hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/AbstractDelegationTokenSecretManager.java

```diff
@@ -771,10 +771,14 @@ protected void logExpireTokens(
     for (TokenIdent ident : expiredTokens) {
       logExpireToken(ident);
       LOG.info("Removing expired token " + formatTokenId(ident));
```

Comment:
`Removing expired token` will be logged even when the token is then not removed.

`removeExpiredStoredToken` could return a boolean indicating whether an token was actually removed. We can use this value to determine whether to print the log line or not.

Reply:
I'm trying to follow the behavior of the other SecretManagers that extend this class, where they log this line and later log an error if anything failed. Some examples on `JHSDelegationTokenSecretManager`, `RMDelegationTokenSecretManager` and `ZKDelegationTokenSecretManager`

I will add a bit more logging on the SQLDTSM to call out if/why the token was not removed. Do you think this will be enough?

Reply:
Yes, I think it will be enough to log the reason the token was not removed. So we'll be able to reconstruct from the logs what happened.